system/doc/reference_manual/processes.xml

```diff
@@ -109,23 +185,503 @@ spawn(Module, Name, Args) -> pid()
+	  when used from a terminated process, but if the NIF is not well
+	  behaved it can still effect other processes. A well behaved dirty
+	  NIF should test if
+	  <seecref marker="erts:erl_nif#enif_is_current_process_alive">the
+	  process it is executing in has exited</seecref>, and if so stop
+	  executing.
+	</p>
+	<p>
+	  In the general case, the heap of a process cannot be removed before
+	  all signals that it needs to send have been sent. Resources held
+	  by heap data are the memory blocks containing the heap, but also
+	  include things referred to from the heap such as off heap binaries,
+	  and resources held via NIF
+	  <seecref marker="erts:erl_nif#resource_objects">resource
+	  objects</seecref> on the heap.
+	</p>
+      </section>
+    </section>
+
+    <section>
+      <marker id="signal-delivery"/>
+      <title>Delivery of Signals</title>
+      <p>
+	The amount of time that passes between the time a signal is sent and
+	the arrival of the signal at the destination is unspecified but
+	positive. If the receiver has terminated, the signal does not arrive,
+	but it can trigger another signal. For example, a <c>link</c> signal
+	sent to a non-existing process triggers an <c>exit</c> signal, which
+	is sent back to where the <c>link</c> signal originated from. When
+	communicating over the distribution, signals can be lost if the
+	distribution channel goes down.
+      </p>
+
+      <p>
+	The only signal ordering guarantee given is the following: if an
+	entity sends multiple signals to the same destination entity, the
+	order is preserved; that is, if <c>A</c> sends a signal <c>S1</c> to
+	<c>B</c>, and later sends signal <c>S2</c> to <c>B</c>, <c>S1</c> is
+	guaranteed not to arrive after <c>S2</c>. Note that <c>S1</c> may,
+	or may not have been lost.
+      </p>
+    </section>
+
+    <section>
+      <marker id="signal-irregularities"/>
+      <title>Irregularities</title>
+      <taglist>
+	<tag>Synchronous Error Checking</tag>
+	<item>
+	  <p>
+	    Some functionality that send signals have synchronous error
+	    checking when sending locally on a node and fail if the receiver
+	    is not present at the time when the signal is sent:
+	  </p>
+	  <list>
+	    <item>
+	      The <seeguide marker="expressions#send">send operator
+	      <c>!</c></seeguide>,
+	      <seemfa marker="erts:erlang#send/2"><c>erlang:send/2,3</c></seemfa>,
+	      BIFs and
+	      <seemfa marker="erts:erlang#send_nosuspend/2"><c>erlang:send_nosuspend/2,3</c></seemfa>
+	      BIFs when the receiver is identified by a name that is
+	      expected to be registered locally.
+	    </item>
+	    <item>
+	      <seemfa marker="erts:erlang#link/1"><c>erlang:link/1</c></seemfa>
+	    </item>
+	    <item>
+	      <seemfa marker="erts:erlang#group_leader/2"><c>erlang:group_leader/2</c></seemfa>
+	    </item>
+	  </list>
+	  <p></p>
+	</item>
+	<tag>Unexpected Behaviour of Exit Signal With Exit Reason <c>normal</c></tag>
```

Comment:
Should we mention this scenario:

```erlang
Eshell V11.2.1  (abort with ^G)
1> process_flag(trap_exit, true).
false
2> spawn_link(fun() -> exit(kill) end).
<0.87.0>
3> flush().
Shell got {'EXIT',<0.87.0>,kill}
ok
```

i.e. that when a link breaks with reason `kill` it is not the same as calling `erlang:exit(Pid,kill)`?